pythonx/UltiSnips/snippet/parsing/base.py

```diff
@@ -57,7 +57,10 @@ def _do_parse(parent, text, allowed_tokens):
             else:
                 klass = token_to_textobject.get(token.__class__, None)
                 if klass is not None:
-                    klass(parent, token)
+                    text_object = klass(parent, token)
+
+                    if isinstance(text_object, TabStop):
```

Comment:
Add a comment that choices token actually materialize into a TabStop instance, please.